reactor-tools/src/main/java/reactor/tools/agent/ReactorDebugAgent.java

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (c) 2019-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor.tools.agent;
+
+import java.lang.instrument.ClassFileTransformer;
+import java.lang.instrument.Instrumentation;
+import java.lang.instrument.UnmodifiableClassException;
+import java.security.ProtectionDomain;
+
+import net.bytebuddy.agent.ByteBuddyAgent;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import reactor.core.publisher.Hooks;
+
+public class ReactorDebugAgent {
+
+	public static void init() {
+		Instrumentation instrumentation = ByteBuddyAgent.install();
```

Comment:
this will throw an exception on Java 8 JVMs (the ones not packed with the JDK, that is. but I don't expect our user running JDK jvms in prod)

Reply:
it does throw indeed, that's a limitation of ByteBuddy.
We could ship a java agent in addition to the self-attachment for the users who don't want to ship Java 8's JDK instead of JRE